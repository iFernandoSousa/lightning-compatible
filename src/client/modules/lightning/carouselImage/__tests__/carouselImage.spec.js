/*
 * Copyright (c) 2019, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: MIT
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
 */

import { createElement } from 'lwc';
import Element from 'c/carouselImage';
import { axe, toHaveNoViolations } from 'jest-axe';

const createCarouselImage = attributes => {
    const element = createElement('c-carousel-image', { is: Element });

    Object.assign(element, attributes);
    document.body.appendChild(element);

    return element;
};

const DEFAULT_ATTRIBUTES = {
    alternativeText: 'This is a card',
    description: 'first card description',
    header: 'First card',
    href: 'https://www.salesforce.com',
    src:
        'https://latest-212.lightningdesignsystem.com/assets/images/carousel/carousel-01.jpg'
};

expect.extend(toHaveNoViolations);

describe('c-carousel-image', () => {
    it('default', () => {
        const element = createCarouselImage(DEFAULT_ATTRIBUTES);

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });

    // eslint-disable-next-line @lwc/lwc/no-async-await
    it('runs accesibility testing on default template', async () => {
        expect.extend(toHaveNoViolations);

        const element = createCarouselImage(DEFAULT_ATTRIBUTES);

        // eslint-disable-next-line @lwc/lwc/no-async-await
        return Promise.resolve().then(async () => {
            // eslint-disable-next-line @lwc/lwc/no-inner-html
            expect(await axe(element.outerHTML)).toHaveNoViolations();
        });
    });
});
