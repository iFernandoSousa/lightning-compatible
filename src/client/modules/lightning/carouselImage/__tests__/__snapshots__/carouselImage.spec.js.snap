// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-carousel-image default 1`] = `
<c-carousel-image
  class="slds-carousel__panel"
  data-handles-touch="true"
>
  #shadow-root(open)
    <div
      aria-hidden="true"
      id="[shadow:guid]"
      role="tabpanel"
    >
      <a
        class="slds-carousel__panel-action slds-text-link_reset"
        href="https://www.salesforce.com"
        tabindex="-1"
      >
        <div
          class="slds-carousel__image"
        >
          <img
            alt="This is a card"
            src="https://latest-212.lightningdesignsystem.com/assets/images/carousel/carousel-01.jpg"
          />
        </div>
        <div
          class="slds-carousel__content"
        >
          <h2
            class="slds-carousel__content-title"
          >
            First card
          </h2>
          <p>
            first card description
          </p>
        </div>
      </a>
    </div>
</c-carousel-image>
`;
